File: Python/Lineares/BuscaSequencial.py
def BuscaSequencial(valor, vetor):
	posicao = -1
	for x in xrange(0, len(vetor)):
		if vetor[x] == valor:
			posicao = x
			break
	return posicao

vetor = [1, 4, 5, 2, 42, 534, 54, 98, 89, 78, 67]
print BuscaSequencial(98, vetor)

File: Ruby/BuscaSequencial.rb
def busca_sequencial(valor, vetor)
  # Procura o valor no vetor sequencialmente
  # Se encontrar retorna o índice, senão -1
  vetor.each_with_index do |elemento, indice|
    return indice if elemento.equal?(valor)
  end
  return -1
end

# Teste Manual
vetor = [1, 4, 5, 2, 42, 34, 54, 98, 89, 78, 67]
puts busca_sequencial(98, vetor)

# Teste Automatizado
require 'minitest/autorun'

class Test < Minitest::Test
  def test_busca_sequencial_quando_nao_encontra_o_valor
    assert_equal busca_sequencial(5, []), -1
  end

  def test_busca_sequencial_quando_encontra_os_valores
    vetor = [1, 4, 5, 2, 42, 34, 54, 98, 89, 78, 67]
    vetor.each_with_index do |elemento, indice|
      assert_equal busca_sequencial(elemento, vetor), indice
    end
  end
end
